### ISSUE TITLE
Black pre-commit and black command line disagree

### ISSUE THREAD

**Issue by justinxzhao** [psf/black#3073]
## Version information

pre-commit version: 2.19.0

```bash
> pre-commit --version
pre-commit 2.19.0
```

pre-commit `black` version: 22.3.0 (https://github.com/ludwig-ai/ludwig/blob/master/.pre-commit-config.yaml)

```yaml
- repo: https://github.com/psf/black
    rev: 22.3.0
    hooks:
      - id: black
        name: black
```

command line `black` version: 22.3.0
```bash
> pip show black
Name: black
Version: 22.3.0
Summary: The uncompromising code formatter.
Home-page: https://github.com/psf/black
Author: Łukasz Langa
Author-email: lukasz@langa.pl
License: MIT
Location: /Users/justinzhao/mambaforge/envs/base38/lib/python3.8/site-packages
Requires: click, mypy-extensions, pathspec, platformdirs, tomli, typing-extensions
```

`pyproject.toml`

```
[tool.isort]
profile = "black"
line_length = 120
force_sort_within_sections = "False"
order_by_type = "False"

[tool.black]
line-length = 120
```

## Example of a formatting discrepancy

`pre-commit run black` formats like so:

```python
                current_learning_rate = (
                    learning_rate_warmup_distributed(
                        current_learning_rate,
                        progress_tracker.epoch,
                        self.learning_rate_warmup_epochs,
                        self.horovod.size(),
                        batcher.step,
                        batcher.steps_per_epoch,
                    )
                    * self.lr_scale_fn(self.horovod.size())
                )
```

`black ludwig/models/trainer.py` prefers this format:

```python
                current_learning_rate = learning_rate_warmup_distributed(
                    current_learning_rate,
                    progress_tracker.epoch,
                    self.learning_rate_warmup_epochs,
                    self.horovod.size(),
                    batcher.step,
                    batcher.steps_per_epoch,
                ) * self.lr_scale_fn(self.horovod.size())
```

AFAICT both `pre-commit black` and `black` are respecting the custom 120-line length limit -- the [file in question](https://github.com/ludwig-ai/ludwig/blob/master/ludwig/models/trainer.py) is huge and this is the only section where the formatting disagrees.

More examples of discrepancies (affects only 8 / 377 files) can be found by running black over the entire repo

```bash
> git clone git@github.com:ludwig-ai/ludwig.git
> pip install black
> pip install pre-commit
> black .
> pre-commit run black
```

## Example CLI output

```bash
> pre-commit run black --all-files
black....................................................................Passed

> black .                         
Skipping .ipynb files as Jupyter dependencies are not installed.
You can fix this by running ``pip install black[jupyter]``
reformatted ludwig/utils/math_utils.py
reformatted ludwig/utils/h3_util.py
reformatted ludwig/modules/mlp_mixer_modules.py
reformatted ludwig/utils/entmax/losses.py
reformatted ludwig/utils/entmax/activations.py
reformatted ludwig/modules/tabnet_modules.py
reformatted ludwig/models/trainer.py
reformatted ludwig/encoders/text_encoders.py
All done! ✨ 🍰 ✨
8 files reformatted, 369 files left unchanged.

> black .
Skipping .ipynb files as Jupyter dependencies are not installed.
You can fix this by running ``pip install black[jupyter]``
All done! ✨ 🍰 ✨
377 files left unchanged.

> pre-commit run black --all-files
black....................................................................Failed
- hook id: black
- files were modified by this hook

reformatted ludwig/utils/math_utils.py
reformatted ludwig/utils/h3_util.py
reformatted ludwig/modules/mlp_mixer_modules.py
reformatted ludwig/utils/entmax/losses.py
reformatted ludwig/utils/entmax/activations.py
reformatted ludwig/modules/tabnet_modules.py
reformatted ludwig/models/trainer.py
reformatted ludwig/encoders/text_encoders.py

All done! ✨ 🍰 ✨
8 files reformatted, 366 files left unchanged.

> pre-commit run black --all-files
black....................................................................Passed
```

**Comment by ichard26** [comment]
Hey thank you for filing this bug report!

Cannot reproduce on Ubuntu 20.04.04 using CPython 3.8.5 ... hmm

```console
~/programming/testing via v3.8.5 took 5s272ms 
128❯ git clone https://github.com/ludwig-ai/ludwig.git
Cloning into 'ludwig'...
remote: Enumerating objects: 26394, done.
remote: Counting objects: 100% (479/479), done.
remote: Compressing objects: 100% (291/291), done.
remote: Total 26394 (delta 278), reused 323 (delta 188), pack-reused 25915
Receiving objects: 100% (26394/26394), 23.87 MiB | 686.00 KiB/s, done.
Resolving deltas: 100% (19620/19620), done.

~/programming/testing via v3.8.5 took 39s512ms 
❯ cd ludwig/

~/programming/testing/ludwig on master via v3.8.5 
❯ python -m virtualenv venv
created virtual environment CPython3.8.5.final.0-64 in 3129ms
  creator CPython3Posix(dest=/home/ichard26/programming/testing/ludwig/venv, clear=False, no_vcs_ignore=False, global=False)
  seeder FromAppData(download=False, pip=bundle, setuptools=bundle, wheel=bundle, via=copy, app_data_dir=/home/ichard26/.local/share/virtualenv)
    added seed packages: pip==22.0.4, setuptools==62.0.0, wheel==0.37.1
  activators BashActivator,CShellActivator,FishActivator,NushellActivator,PowerShellActivator,PythonActivator

~/programming/testing/ludwig on master via v3.8.5 took 3s835ms 
❯ actenv
.: Error encountered while sourcing file “env/bin/activate.fish”:
.: No such file or directory
... but venv did work!

~/programming/testing/ludwig on master via v3.8.5 (venv) 
❯ pip install black
Collecting black
  Using cached black-22.3.0-cp38-cp38-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (1.5 MB)
Collecting typing-extensions>=3.10.0.0
  Downloading typing_extensions-4.2.0-py3-none-any.whl (24 kB)
Collecting platformdirs>=2
  Downloading platformdirs-2.5.2-py3-none-any.whl (14 kB)
Collecting click>=8.0.0
  Downloading click-8.1.3-py3-none-any.whl (96 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 96.6/96.6 KB 301.3 kB/s eta 0:00:00
Collecting tomli>=1.1.0
  Using cached tomli-2.0.1-py3-none-any.whl (12 kB)
Collecting pathspec>=0.9.0
  Using cached pathspec-0.9.0-py2.py3-none-any.whl (31 kB)
Collecting mypy-extensions>=0.4.3
  Using cached mypy_extensions-0.4.3-py2.py3-none-any.whl (4.5 kB)
Installing collected packages: mypy-extensions, typing-extensions, tomli, platformdirs, pathspec, click, black
Successfully installed black-22.3.0 click-8.1.3 mypy-extensions-0.4.3 pathspec-0.9.0 platformdirs-2.5.2 tomli-2.0.1 typing-extensions-4.2.0
WARNING: You are using pip version 22.0.4; however, version 22.1 is available.
You should consider upgrading via the '/home/ichard26/programming/testing/ludwig/venv/bin/python -m pip install --upgrade pip' command.

~/programming/testing/ludwig on master via v3.8.5 (venv) took 7s115ms 
❯ dropblackcache -y
found cache directory used by Black 0.1.dev1+gae2c075
found cache directory used by Black 22.3.0
found cache directory used by Black 22.3.1.dev31+gbd7373c
found cache directory used by Black 22.3.1.dev24+g7f7673d
found cache directory used by Black 0.1.1a2
found cache directory used by Black 22.1.0
successfully dropped cache directory used by Black 0.1.dev1+gae2c075
successfully dropped cache directory used by Black 22.3.0
successfully dropped cache directory used by Black 22.3.1.dev31+gbd7373c
successfully dropped cache directory used by Black 22.3.1.dev24+g7f7673d
successfully dropped cache directory used by Black 0.1.1a2
successfully dropped cache directory used by Black 22.1.0

~/programming/testing/ludwig on master via v3.8.5 (venv) took 291ms 
❯ black . --check --diff --color
Skipping .ipynb files as Jupyter dependencies are not installed.
You can fix this by running ``pip install black[jupyter]``
All done! ✨ 🍰 ✨
387 files would be left unchanged.

~/programming/testing/ludwig on master via v3.8.5 (venv) took 14s202ms 
❯ pre-commit run black --all-files -v
Format code..............................................................Passed
- hook id: black
- duration: 29.84s

All done! ✨ 🍰 ✨
387 files left unchanged.
```

What OS and version of Python are you using? I'm wondering if mypyc is causing trouble and it's specific to one or a few wheels (pre-commit doesn't use the compiled wheels on PyPI). Also it may be worth trying `pre-commit clean` to remove all hook environments (sadly pre-commit doesn't want let us specify a single hook >.<)

**Comment by justinxzhao** [comment]
Wow! Fast reply -- thanks :)

Glad to hear that it's not reproducing on your end, which means it's likely just an environment issue.

I'm still seeing the same disagreement with `pre-commit clean`.

My python version:

```bash
> python --version
Python 3.8.13
```

My OS: 
Mac OS Monterey 12.2

I did notice that while `pip show black` prints version `22.3.0`, when I run `black --version`, I get a different version (that's also not compiled):

```bash
> black --version
black, 21.12b0 (compiled: no)
```

Could that be the culprit?

**Comment by andresito00** [comment]
@justinxzhao I thought I was seeing something like this, but later realized that I had more than 1 *.toml files. 

In my case, I narrowed down the behavior I was seeing to a "one" vs. "many" situation (also exposed by pre-commit).

My observation is that `black` will find the "outermost" .toml or .git project configuration, with respect to the directory structure, and use _that_ for a given set of files.

In a situation where only one file is provided, it is possible that an "inner" .toml is selected and used to apply formatting.

I saw this by running `black FILE_PATH --verbose` and `black . --verbose`.

**Comment by ichard26** [comment]
> Could that be the culprit?

Yeah, we changed the formatting style quite a bit between 21.12b0 and 22.1.0 around the magic trailing comma. I suspect that's it. Whether Black is compiled _shouldn't_ change formatting, but it always could. Don't think it's that here though.

> Glad to hear that it's not reproducing on your end, which means it's likely just an environment issue.

I agree so I'll be closing this. Thanks for reporting either way!

--- 

@andresito00 if you have a suggestion in regards to your issue, please file a new issue since it seems unrelated. Not sure if it was just meant as an alternative explanation or also wanted a solution. Thanks!
